### PR TITLE
Discard key share data when fingerprinting ClientHello

### PR DESCRIPTION
**Pull request by hwh33**
I noticed that we are keeping key share data when fingerprinting ClientHellos.  I think this could create an issue as key share data should usually be created uniquely for each connection.  For reference, see the logic here:

https://github.com/getlantern/utls/blob/a20f07cc4a907e9bcfc7f9eeb962ba5bc91e1d9e/u_parrots.go#L657-L680

This code block is responsible for populating the actual key share data sent on the wire.  If the spec already contains key share data, no new data will be created.  Thus all hellos will share identical key share data.  I'm not well-versed enough to know the full security ramifications, but it doesn't seem good.  At the least, it presents an avenue for fingerprinting the hellos.

@max-b how is the test data generated?  It looks like I will need to re-generate expected test values.

### CODE REVIEW

**Comment by max-b** [comment]
Good catch! The tests are a *bear* to update. You have to build your own version of `openssl`: https://github.com/getlantern/utls/blob/6f3f4a064fb6bde95a765c84f845ebb41abd67e2/handshake_test.go#L75-L85

I've already gone through that, so let me do that now...

**Comment by max-b** [comment]
:tada: tests passing now :grin:

**Comment by hwh33** [comment]
Wow, thanks for updating the tests @max-b!  That was nice of you.  I'll go ahead and merge this now.

File: u_parrots.go [hunk 1]
```diff
@@ -1156,6 +1156,11 @@ func FingerprintClientHello(data []byte) (*ClientHelloSpec, error) {
 					return nil, errors.New("unable to read key share extension data")
 				}
 				ks.Group = CurveID(unGREASEUint16(group))
+				// if not GREASE, key share data will be discarded
+				// as it should be generated per connection
+				if ks.Group != GREASE_PLACEHOLDER {
+					ks.Data = nil
+				}
```
Comment (hwh33) [inline u_parrots.go:1163]: Oh nice, didn't realize we'd need to keep this for GREASE

**Pull request merged**